### DISCUSSION TITLE
TS missing fullPath property on .upload() return type

### DISCUSSION THREAD

**Discussion by Jervx** [supabase/supabase#21176] [Questions] [answered]
I have this code

```
let uploadResult = await createClient()
  .storage.from(bucket)
  .upload(`${fileName}.${fileExtension}`, imageFile, {
      cacheControl: "3600",
      upsert,
});
```
and the type is only showing the { path : string }
![image](https://github.com/supabase/supabase/assets/48374007/7587a84d-3cc6-41b5-b297-c7f20698939b)

but on console.log after uploading a file, it also shows fullPath. Is this intentional to not included the fullPath on return type?

![image](https://github.com/supabase/supabase/assets/48374007/ce82338b-aec2-4949-804d-a86e2f1b8e0b)

**Comment by GaryAustin1** [comment]
Supabase-js does not return either the id or the fullPath.
![image](https://github.com/supabase/supabase/assets/54564956/280923a1-3a02-4ed3-b8c6-164732e21f06)

**Reply by IHIutch** [reply]
I too am getting `id` and `fullPath`. Can you elaborate? I'm using v2.39.3

<img width="434" alt="image" src="https://github.com/supabase/supabase/assets/20825047/133f54b1-e6b4-4ef7-8b6a-03beb3f7ee9c">
 
 
 
 Looks like `upload` calls `uploadOrUpdate` which does return `id` and `fullPath`: https://github.com/supabase/storage-js/blob/68a61973a4bdaa36a41014679108add89717bb38/src/packages/StorageFileApi.ts#L64-L135

**Reply by GaryAustin1** [reply]
I am now getting the same results with latest supabase-js 2.39.7.  I don't know the version I was running, but normally it is current for the time.  So a bit puzzled.
![image](https://github.com/supabase/supabase/assets/54564956/21b09138-3b3b-46cd-af38-4948c33ee271)

**Reply by GaryAustin1** [reply, marked as answer]
Looks like it was fixed a few weeks ago, so not sure when actually released.
![image](https://github.com/supabase/supabase/assets/54564956/09969493-4888-4529-b7ec-402ab9257d2a)
![image](https://github.com/supabase/supabase/assets/54564956/ca8824ce-bd0d-480c-9f7b-153acec6b9b9)
